# Análise de faculdades globais

In [512]:
import pandas as pd
import numpy as np
import seaborn as sns


# Tratamento dos dados

## Criando dataset

In [513]:
dataset = pd.read_csv('College Data.csv')
pd.set_option('display.max_columns', None)

## Tratamento do dataset
Aqui renomearemos colunas, formataremos dados para melhor compreensão e removeremos erros contidos no dataset.

In [514]:
renomear_colunas = ['Faculdade ID', 'Pais', 'Total_Estudantes', 'Estudantes_Masculinos', 'Estudantes_Femininas', 'CGPA',
       'Renda_Familiar_Anual', 'Curso', 'Esportes', 'Pesquisas_Publicadas', 'Taxa_Formados_Empregados', 'Número_Professores']
dataset.columns = renomear_colunas

dataset['CGPA'] = dataset["CGPA"].round(1)
dataset['Taxa_Formados_Empregados'] = pd.to_numeric(dataset['Taxa_Formados_Empregados'], errors='coerce')

dataset['Estudantes_Femininas'] = dataset['Estudantes_Femininas'].abs()


## Visualizando o dataset

In [515]:
dataset

,Faculdade ID,Pais,Total_Estudantes,Estudantes_Masculinos,Estudantes_Femininas,CGPA,Renda_Familiar_Anual,Curso,Esportes,Pesquisas_Publicadas,Taxa_Formados_Empregados,Número_Professores
0,College 52,UK,27002,2593,11644,10.0,3574250,Physics,Swimming,39,100.00,655
1,College 93,Switzerland,11913,9392,2521,7.0,762556,Chemical Engineering,Athletics,31,68.84,256
2,College 15,Japan,15644,5475,4388,9.3,1795845,Business Administration,Volleyball,37,91.99,555
3,College 72,Singapore,13239,7363,5876,8.7,1222223,Physics,Tennis,48,97.98,455
4,College 61,Canada,10236,13339,3103,8.6,1921133,Biotechnology,Athletics,34,83.56,310
...,...,...,...,...,...,...,...,...,...,...,...,...
49995,College 93,India,14676,10771,3905,8.3,3151197,Civil Engineering,Cricket,20,65.61,596
49996,College 48,Japan,21908,10682,6820,8.5,3060006,Physics,Chess,9,99.15,383
49997,College 67,South Africa,17263,11035,6228,6.0,1519563,Chemical Engineering,Athletics,35,78.28,671
49998,College 54,USA,11371,10233,1138,7.7,2429788,Mechanical Engineering,Swimming,11,88.46,164


### Informações do dataset
Aqui veremos quantas linhas não nulas o dataset possuí e o tipo de dados de cada coluna

In [516]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 12 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Faculdade ID              50000 non-null  object 
 1   Pais                      50000 non-null  object 
 2   Total_Estudantes          50000 non-null  int64  
 3   Estudantes_Masculinos     50000 non-null  int64  
 4   Estudantes_Femininas      50000 non-null  int64  
 5   CGPA                      50000 non-null  float64
 6   Renda_Familiar_Anual      50000 non-null  int64  
 7   Curso                     50000 non-null  object 
 8   Esportes                  50000 non-null  object 
 9   Pesquisas_Publicadas      50000 non-null  int64  
 10  Taxa_Formados_Empregados  50000 non-null  float64
 11  Número_Professores        50000 non-null  int64  
dtypes: float64(2), int64(6), object(4)
memory usage: 4.6+ MB


### Descrição dos dados
Aqui temos uma breve descrição estatistica dos dados númericos, como média de valor das colunas, desvio padrão, seus valores mínimos e máximos.

In [517]:
dataset.describe().round(1)

,Total_Estudantes,Estudantes_Masculinos,Estudantes_Femininas,CGPA,Renda_Familiar_Anual,Pesquisas_Publicadas,Taxa_Formados_Empregados,Número_Professores
count,50000.0,50000.0,50000.0,50000.0,50000.0,50000.0,50000.0,50000.0
mean,15522.4,7747.1,5721.7,8.7,2611605.6,24.5,82.3,428.9
std,8366.8,4188.4,3824.0,1.1,1387889.0,14.5,11.5,154.7
min,1001.0,500.0,0.0,4.0,200004.0,0.0,60.0,62.0
25%,8272.8,4111.0,2487.0,8.0,1408785.5,12.0,72.4,315.0
50%,15506.5,7756.0,5112.0,8.8,2620902.5,24.0,82.5,428.0
75%,22768.2,11360.0,8504.0,9.7,3811208.5,37.0,92.5,544.0
max,29999.0,14999.0,14999.0,10.0,4999988.0,49.0,100.0,796.0


# Análises Iniciais

In [ ]:
dataset[dataset['Faculdade ID'] == 'College 61']

## Analisando quantidade de estudante por país

### Quantidade geral de estudante por país

In [518]:
qtd_estud_pais = dataset.groupby('Pais')['Total_Estudantes'].sum().reset_index()
qtd_estud_pais.sort_values(by='Total_Estudantes',ascending=False)

,Pais,Total_Estudantes
2,China,78785524
6,South Africa,78611487
7,Switzerland,78233472
5,Singapore,77965255
0,Australia,77368139
9,USA,77299951
4,Japan,77166258
1,Canada,77057301
3,India,76900811
8,UK,76729674


### Quantidade de estudante por gênero

In [519]:
qtd_estud_masc_pais = dataset.groupby('Pais')['Estudantes_Masculinos'].sum().reset_index()
qtd_estud_masc_pais.sort_values(by='Estudantes_Masculinos',ascending=False)

,Pais,Estudantes_Masculinos
6,South Africa,39302765
2,China,39101368
0,Australia,39020306
9,USA,38865321
7,Switzerland,38792394
5,Singapore,38670432
1,Canada,38596759
4,Japan,38573411
3,India,38468236
8,UK,37962103


In [520]:
qtd_estud_masc_pais = dataset.groupby('Pais')['Estudantes_Femininas'].sum().reset_index()
qtd_estud_masc_pais.sort_values(by='Estudantes_Femininas',ascending=False)

,Pais,Estudantes_Femininas
2,China,29516477
9,USA,28842841
0,Australia,28832090
7,Switzerland,28796949
1,Canada,28733336
5,Singapore,28447927
4,Japan,28399963
6,South Africa,28344085
3,India,28161373
8,UK,28008096


### Porcentagem de estudantes por gênero

In [521]:
porcentagem = (dataset.groupby('Pais')['Estudantes_Masculinos'].sum() / dataset.groupby('Pais')['Total_Estudantes'].sum()) * 100
porcentagem = porcentagem.round(2).reset_index()
display(porcentagem)

,Pais,0
0,Australia,50.43
1,Canada,50.09
2,China,49.63
3,India,50.02
4,Japan,49.99
5,Singapore,49.60
6,South Africa,50.00
7,Switzerland,49.59
8,UK,49.48
9,USA,50.28


In [522]:
porcentagem = (dataset.groupby('Pais')['Estudantes_Femininas'].sum() / dataset.groupby('Pais')['Total_Estudantes'].sum()) * 100
porcentagem = porcentagem.round(2).reset_index()
display(porcentagem)

,Pais,0
0,Australia,37.27
1,Canada,37.29
2,China,37.46
3,India,36.62
4,Japan,36.80
5,Singapore,36.49
6,South Africa,36.06
7,Switzerland,36.81
8,UK,36.50
9,USA,37.31


Opa, podemos notar que ao somar as porcetagens de estudantes por gênero não conseguimos chegar a 100% da quantidade de estudantes daquele país, vamos averiguar.

#### Hipótese
acredito que há universidades em que a quantidade total de estudantes não é o mesmo valor que o da soma entre a quantidade de estudantes masculinos e femininos, fazendo a universidade ter um certo 'gap' que não é preenchido

In [523]:
dataset[(dataset['Estudantes_Femininas'] + dataset['Estudantes_Masculinos']) != dataset['Total_Estudantes']]

,Faculdade ID,Pais,Total_Estudantes,Estudantes_Masculinos,Estudantes_Femininas,CGPA,Renda_Familiar_Anual,Curso,Esportes,Pesquisas_Publicadas,Taxa_Formados_Empregados,Número_Professores
0,College 52,UK,27002,2593,11644,10.0,3574250,Physics,Swimming,39,100.00,655
2,College 15,Japan,15644,5475,4388,9.3,1795845,Business Administration,Volleyball,37,91.99,555
4,College 61,Canada,10236,13339,3103,8.6,1921133,Biotechnology,Athletics,34,83.56,310
5,College 21,USA,22804,4373,2757,9.1,1885655,Chemical Engineering,Basketball,30,85.50,439
6,College 83,UK,8135,11111,2976,6.9,1191708,Biotechnology,Volleyball,21,76.13,282
...,...,...,...,...,...,...,...,...,...,...,...,...
49989,College 73,UK,27394,4720,8287,8.2,1731532,Computer Science,Chess,31,84.72,604
49990,College 56,Switzerland,9287,6309,912,10.0,4018482,Economics,Cricket,24,75.21,556
49991,College 6,Australia,22435,10439,10369,7.9,1435003,Civil Engineering,Basketball,12,73.44,668
49992,College 28,Australia,15347,5114,6299,10.0,1916342,Computer Science,Cricket,15,85.33,360


Há 36566 universidades em que isso acontece, essa imprecissão nos dados não nos será útil então filtraremos o dataset a fim de termos mais precissão em nossos dados.

In [524]:
dataset = dataset[(dataset['Estudantes_Femininas'] + dataset['Estudantes_Masculinos']) == dataset['Total_Estudantes']]

In [525]:
qtd_estud_pais = dataset.groupby('Pais')['Total_Estudantes'].sum().reset_index()
qtd_estud_pais.sort_values(by='Total_Estudantes',ascending=False)

,Pais,Total_Estudantes
0,Australia,17998756
9,USA,17717766
6,South Africa,17585553
4,Japan,17318786
3,India,17149112
8,UK,17039594
7,Switzerland,17019677
1,Canada,16966030
5,Singapore,16711993
2,China,16556804


podemos notar a queda drástica no número de estudantes por país, sendo que anteriormente tínhamos 78785524 estudantes na china e agora só temos 16556804, tendo uma perca de cerga de 62228720 estudantes.